metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1277,15 +1277,20 @@ class MetalsLanguageServer(
 
   private def indexWorkspaceSources(): Unit = {
     for {
-      (sourceDirectory, targets) <- buildTargets.sourceDirectoriesToBuildTargets
-      if sourceDirectory.isDirectory
-      source <- ListFiles(sourceDirectory)
+      (sourceItem, targets) <- buildTargets.sourceItemsToBuildTargets
+      sources = if (sourceItem.isDirectory) ListFiles(sourceItem)
+      else List(sourceItem)
+      source <- sources
       if source.isScalaOrJava
     } {
       targets.asScala.foreach { target =>
         buildTargets.linkSourceFile(target, source)
       }
-      indexSourceFile(source, Some(sourceDirectory))
+      val sourceDirectory =
```

Comment:
What do you think about:

```scala
if(sourceItem.isDirectory) {
  indexSourceFile(source, Some(sourceItem))
} else if (source.exists) { //here should be a comment what does it mean when a source exists but a sourceItem is not a directory
  indexSourceFile(source, None)
}
```

same amount of lines but at the same time more expressive